### PR TITLE
Create and modify release bundles

### PR DESCRIPTION
**Pull request by yahavi**
- [x] I signed [JFrog's CLA](https://secure.echosign.com/public/hostedForm?formid=5IYKLZ2RXB543N).
- [x] All [tests](https://github.com/jfrog/jfrog-cli#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [x] This pull request is on the dev branch.
- [x] I used gofmt for formatting the code before submitting the pull request.
-----

Add the following commands:
* jfrog rt release-bundle-create (rbc)
* jfrog rt release-bundle-update (rbu)
* jfrog rt release-bundle-sign (rbs)
* jfrog rt release-bundle-distribute (rbd)
* jfrog rt release-bundle-delete (rbdel)

Related to:
https://github.com/jfrog/jfrog-client-go/pull/126
jfrog/gocmd#25

### CODE REVIEW

File: artifactory/cli.go [hunk 1]
```diff
@@ -2737,13 +2908,130 @@ func buildDiscardCmd(c *cli.Context) error {
...
+}
+
+func releaseBundleDistributeCmd(c *cli.Context) error {
+	if c.NArg() != 2 {
+		return cliutils.PrintHelpAndReturnError("Wrong number of arguments.", c)
+	}
+	var distributionRules *spec.DistributionRules
+	if c.IsSet("distribution-rules") {
+		if c.IsSet("site-name") || c.IsSet("city-name") || c.IsSet("country-code") {
+			return cliutils.PrintHelpAndReturnError("flag --distribution-rules can't be used with --site-name, --city-name or --country-code", c)
```
Comment (eyalbe4) [inline artifactory/cli.go]: flag --distribution-rules ... --> The --dist-rules option...

File: artifactory/cli.go [hunk 2]
```diff
@@ -2737,13 +2908,130 @@ func buildDiscardCmd(c *cli.Context) error {
...
+}
+
+func releaseBundleDeleteCmd(c *cli.Context) error {
+	if c.NArg() != 2 {
+		return cliutils.PrintHelpAndReturnError("Wrong number of arguments.", c)
+	}
+	var distributionRules *spec.DistributionRules
+	if c.IsSet("distribution-rules") {
+		if c.IsSet("site-name") || c.IsSet("city-name") || c.IsSet("country-code") {
+			return cliutils.PrintHelpAndReturnError("flag --distribution-rules can't be used with --site-name, --city-name or --country-code", c)
```
Comment (eyalbe4) [inline artifactory/cli.go]: flag --distribution-rules... --> The --dist-rules option...

File: docs/artifactory/releasebundlecreate/help.go [hunk 3]
```diff
@@ -0,0 +1,16 @@
+package releasebundlecreate
+
+const Description = "Create a release bundle."
+
+var Usage = []string{"jfrog rt rbc [command options] <bundle name> <bundle version> <pattern>",
```
Comment (eyalbe4) [inline docs/artifactory/releasebundlecreate/help.go]: Let's use here "release bundle name"  and "release bundle version"

File: docs/artifactory/releasebundlecreate/help.go [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+	"jfrog rt rbc --spec=<File Spec path> [command options] <bundle name> <bundle version>"}
+
+const Arguments string = `	bundle name
```
Comment (eyalbe4) [inline docs/artifactory/releasebundlecreate/help.go]: bundle name --> release bundle name

File: docs/artifactory/releasebundlecreate/help.go [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+		Build name.
```
Comment (eyalbe4) [inline docs/artifactory/releasebundlecreate/help.go]: Build name. --> The name of the release bundle.

File: docs/artifactory/releasebundlecreate/help.go [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+
+	bundle version
+		Bundle version.
```
Comment (eyalbe4) [inline docs/artifactory/releasebundlecreate/help.go]: Bundle version. --> The release bundle version.

File: docs/artifactory/releasebundlecreate/help.go [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+
+const Description = "Create a release bundle."
+
+var Usage = []string{"jfrog rt rbc [command options] <bundle name> <bundle version> <pattern>",
+	"jfrog rt rbc --spec=<File Spec path> [command options] <bundle name> <bundle version>"}
+
+const Arguments string = `	bundle name
+		Build name.
+
+	bundle version
```
Comment (eyalbe4) [inline docs/artifactory/releasebundlecreate/help.go]: bundle version --> release bundle version

File: docs/artifactory/releasebundlecreate/help.go [hunk 8]
```diff
@@ -0,0 +1,16 @@
+package releasebundlecreate
+
```
Comment (eyalbe4) [inline docs/artifactory/releasebundlecreate/help.go]: Let's implement the changes suggested for this file, in all other release bundle help files.

File: inttestutils/distribution.go [hunk 9]
```diff
@@ -109,66 +113,49 @@ func DeleteGpgKeys(artHttpDetails httputils.HttpClientDetails) {
...
 }
 
-// Create the AQL for the release bundle creation
-func createAqlForCreateBundle(bundleName, bundleVersion string, triples []RepoPathName) string {
-	innerQueryPattern := "{\\\"$or\\\":["
-	for i, triple := range triples {
-		innerQueryPattern += fmt.Sprintf(repoPathNameAqlPattern, triple.Repo, triple.Path, triple.Name)
-		if i+1 < len(triples) {
-			innerQueryPattern += ","
+// Return true if the release bundle is exist locally on distribution
```
Comment (eyalbe4) [inline inttestutils/distribution.go]: ...bundle is exist... --> ...bundle exists...

File: artifactory/commands/distribution/distributebundle.go [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+)
+
+type DistributeBundleCommand struct {
+	rtDetails               *config.ArtifactoryDetails
+	distributeBundlesParams services.DistributionParams
+	distributionRules       *spec.DistributionRules
+	dryRun                  bool
+}
+
+func NewReleaseBundleDistributeCommand() *DistributeBundleCommand {
```
[the remark this answers is not in the document]
Comment (yahavi) [reply artifactory/commands/distribution/distributebundle.go]: But in other release-bundle and build commands the `new` function is
```
NewReleaseBundle<action>Command
NewBuild<action>Command
```

File: artifactory/commands/distribution/deletebundle.go [hunk 11]
```diff
@@ -0,0 +1,129 @@
...
+
+func (db *DeleteBundleCommand) RtDetails() (*config.ArtifactoryDetails, error) {
+	return db.rtDetails, nil
+}
+
+func (db *DeleteBundleCommand) CommandName() string {
+	return "rt_delete_bundle"
+}
+
+// Return true iff there are no distribution rules
```
[the remark this answers is not in the document]
Comment (yahavi) [reply artifactory/commands/distribution/deletebundle.go]: Iff = if and only if:
https://www.wikiwand.com/en/If_and_only_if

**Pull request merged**